hack/tools/release/internal/update_providers/provider_issues.go

```diff
@@ -0,0 +1,352 @@
+		if resp.StatusCode != http.StatusCreated {
+			fmt.Println("Failed to create issue for the repository:", repo, "Status:", resp.Status)
+			issuedFailed = append(issuedFailed, repo)
+		} else {
+			responseBody, err := io.ReadAll(resp.Body)
+			if err != nil {
+				fmt.Printf("Failed to read response body: %s\n", err)
+				err := resp.Body.Close()
+				if err != nil {
+					fmt.Printf("Failed to close response body: %s\n", err)
+				}
+				os.Exit(1)
+			}
+
+			var issueResponse IssueResponse
+			err = json.Unmarshal(responseBody, &issueResponse)
+			if err != nil {
+				fmt.Printf("Failed to unmarshal issue response: %s\n", err)
+				err := resp.Body.Close()
+				if err != nil {
+					fmt.Printf("Failed to close response body: %s\n", err)
+				}
+				os.Exit(1)
+			}
+
+			fmt.Println("Issue created for repository:", repo)
+			issuesCreated = append(issuesCreated, issueResponse.HTMLURL)
+		}
+		err = resp.Body.Close()
+		if err != nil {
+			fmt.Printf("Failed to close response body: %s\n", err)
+		}
+	}
+
+	if len(issuesCreated) > 0 || len(issuedFailed) > 0 {
+		fmt.Printf("\n")
+		fmt.Println("###############################################")
+		fmt.Println("Summary:")
+		fmt.Println("###############################################")
+	}
+
+	if len(issuesCreated) > 0 {
+		fmt.Printf("\n")
+		fmt.Println("List of issues created:")
+		fmt.Println("-", strings.Join(issuesCreated, "\n- "))
+	}
+
+	if len(issuedFailed) > 0 {
+		fmt.Printf("\n")
+		fmt.Println("List of issues failed to create:")
+		fmt.Println("-", strings.Join(issuedFailed, "\n- "))
+	}
+}
+
+// continueOrAbort asks the user to continue or abort the script.
+func continueOrAbort() {
+	fmt.Println("Continue? (y/n)")
+	var response string
+	_, err := fmt.Scanln(&response)
+	if err != nil {
+		fmt.Printf("Failed to read response: %s\n", err)
+		os.Exit(1)
+	}
+	if response != "y" {
+		fmt.Println("Aborting...")
+		os.Exit(0)
+	}
+}
+
+// getReleaseDetails returns the release details from the environment variables.
+func getReleaseDetails() releaseDetails {
+	releaseSemVer, keySet := os.LookupEnv("RELEASE_SEMVER")
+	if !keySet || releaseSemVer == "" {
+		fmt.Println("RELEASE_SEMVER is a required environmental variable.")
+		fmt.Println("Refer to README.md in folder for more information.")
+		os.Exit(1)
+	}
+
+	match, err := regexp.Match("\\d\\.\\d", []byte(releaseSemVer))
+	if err != nil || !match {
+		fmt.Println("RELEASE_SEMVER must be in format `\\d\\.\\d` e.g. 1.5")
+		os.Exit(1)
+	}
+
+	releaseDate, keySet := os.LookupEnv("RELEASE_DATE")
+	if !keySet || releaseDate == "" {
+		fmt.Println("RELEASE_DATE is a required environmental variable.")
+		fmt.Println("Refer to README.md in folder for more information.")
+		os.Exit(1)
+	}
+
+	formattedReleaseDate, err := formatDate(releaseDate)
+	if err != nil {
+		fmt.Println("Unable to parse the date.", err)
+		fmt.Println("Refer to README.md in folder for more information.")
+	}
+
+	releaseTag := fmt.Sprintf("v%s%s", releaseSemVer, ".0")
```

Comment:
This can be removed when changing the release version format.

Reply:
Acknowledged